staging/src/k8s.io/cloud-provider/app/core.go

```diff
@@ -104,15 +104,15 @@ func startServiceController(ctx *config.CompletedConfig, cloud cloudprovider.Int
 }
 
 func startRouteController(ctx *config.CompletedConfig, cloud cloudprovider.Interface, stopCh <-chan struct{}) (http.Handler, bool, error) {
-	if !ctx.ComponentConfig.KubeCloudShared.AllocateNodeCIDRs || !ctx.ComponentConfig.KubeCloudShared.ConfigureCloudRoutes {
-		klog.Infof("Will not configure cloud provider routes for allocate-node-cidrs: %v, configure-cloud-routes: %v.", ctx.ComponentConfig.KubeCloudShared.AllocateNodeCIDRs, ctx.ComponentConfig.KubeCloudShared.ConfigureCloudRoutes)
+	if !ctx.ComponentConfig.KubeCloudShared.ConfigureCloudRoutes {
```

Comment:
Contemplating if this is considered a breaking change since `AllocateNodeCIDRs` is false by default and `ConfigureCloudRoutes` is true by default. This means if you previously set neither of these flags, this check is now passing. I think it isn't because:
1. the cloud provider still needs to implement `Routes()` to enable the controller.
2. even if the cloud provider implements `Routes()`, routes controller itself is no-op if `node.spec.podCIDR` is not set. If `node.spec.podCIDR` is set, it means `--allocate-node-cidr` is enabled on kube-controller-manager, which means we should be running this controller.

Reply:
@cheftako would still like your thoughts/opinions on this

Reply:
My worry would be the managed vs unmanaged distinction. Take the GCP Cloud Provider; it has to implement Routes() as the same cloud provider impl is used in both managed (GKE) and unmanaged releases. The managed is not an issue as we have a dedicated team of experts keeping it working. 

For unmanaged releases allocate-node-cidrs defaults to true and enable-ip-aliases to false (i.e. configure-cloud-routes to true). Given that allocate-node-cidrs defaults to true and routes() is implemented, the unmanaged instance would have failed this case but we're saved by configure-cloud-routes being true. Not sure I'm sanguine about being safe in the other cases where we may want different behaviors between managed and unmanaged kubernetes deployments.